Add tutorial notes

Anbranin opened the pull request:
Make em do tutorials

werebus commented on config/qualifications.yml:
```diff
@@ -32,3 +33,85 @@ description: >
+  Pick one from this list, or if you find one you like better, get approval from your supervisor.
+  - [Learn enough Ruby to be dangerous](https://www.learnenough.com/ruby-tutorial/hello_world)
+  - [Ruby Koans](http://rubykoans.com/)
+  rubymonk(maybe), learn enough Ruby to be dangerous, ruby koans
```
This looks like perhaps a note to yourself?

Anbranin replied:
sure is buddy

werebus commented on config/qualifications.yml:
```diff
@@ -32,3 +33,85 @@ description: >
+  - [Ruby Koans](http://rubykoans.com/)
+  rubymonk(maybe), learn enough Ruby to be dangerous, ruby koans
+---
+title: Complete a git tutorial (optional)
```
I think I'd put these before the Rails tutorials. Order doesn't matter _too_ much_ but the issues will be created in this order.

werebus replied:
Also the command line?

werebus commented on config/qualifications.yml:
```diff
@@ -32,3 +33,85 @@ description: >
+  Use [this tutorial](https://www.learnenough.com/command-line-tutorial/basics)
+  or another approved tutorial to brush up on your basics.
+---
+title: Demonstrate understanding of Ruby core methods / Rails object extensions
```
I feel like I've already reviewed this? #23 perhaps?

Anbranin replied:
I think this happens when I don't manually update the diff.

Anbranin replied:
Still shows up as new

werebus replied:
Yeah, I missed that #23 wasn't a PR into master.

Anbranin commented:
Why is my build broken

werebus commented:
Man, am I tired of Bundler bullshit on Travis.

I added (a0f4b937c4aa3f92eb87b71637c7769065af27ee) steps to update Bundler in CI, but since then Bundler 2.0.2 was released, I guess?

And it's like, "hey, hey, hey, you said you need bundler **2.0.1**. What R U doing?"

Anbranin commented:
Well so I can change the `.travis.yml` file to have bundler 2.0.1 OR I can bundle update --bundler? I think that will also work. Will take longer. Also we'd have to keep the BUNDLED WITH updated as well.

Anbranin commented:
@werebus I think I've addressed all your issues, if you'd like to take another look. Also, wondering if I did the right thing for Travis.

werebus commented:
First, I think #24 (just opened) is a bit better fix for the Bundler version issue.

Second, I'm not 100% sure I agree with, "all of the issues are technically optional if one can demonstrate competence." For "go off and do a thing" type issues, you aren't really asking for that demonstration.

Anbranin commented:
OK, what do you suggest about the go-off-and-do-a-thing type issues? I'm now leaning towards making none of them optional. My idea is that to close the issue, include a link to which tutorial or whatnot you did. If you're more competent, you can do some advanced reading. Just something. I could include suggestions.

LiamBrandt commented on config/qualifications.yml:
```diff
@@ -33,3 +34,87 @@ description: >
+description: >
+  Show that you understand how to properly use Git by writing commits that
+  generally relate to atomic features or thoughts. Make sure you know how to
+  resolve a merge conflict correctly.
```
How are you supposed to show that you know how to resolve a merge conflict correctly? Just ask somebody if what you are doing is correct?

LiamBrandt commented on config/qualifications.yml:
```diff
@@ -33,3 +34,87 @@ description: >
+title: display a good understanding of how and why to write tests
+description: >
+  Demonstrate knowledge of basic model/controller/system test structure.
+  Demonstrate understanding of mocks/stubs and what purpose they serve.
```
How does a trainee demonstrate to themselves that they have this knowledge or understanding? Should they write tests for some part of an app?

Anbranin replied:
Yes. "demonstrate" in most cases means "write code that uses"

Anbranin commented:
@werebus, I'd _prefer_ not to without your express approval. I think at this point I've addressed all the comments.
I'd like to actually use this for the current batch of trainees because I think real life feedback is desirable at this point.
